mypy/checker.py

```diff
@@ -72,19 +73,31 @@
 
 DEFAULT_LAST_PASS = 1  # type: Final  # Pass numbers start at 0
 
+DeferredNodeType = Union[FuncDef, LambdaExpr, OverloadedFuncDef, Decorator]
+FineDeferredNodeType = Union[FuncDef, MypyFile, OverloadedFuncDef]
```

Comment:
Style nit: I'd prefer to call this `FineGrainedDeferredNodeType` or something. 

We don't use "fine" as a shorted version of "fine-grained" elsewhere and it can be confusing.